### PR TITLE
Adds the ``execution_environment`` attribute to the Galaxy V3 CollectionVersion APIs

### PR DESCRIPTION
**Pull request by bmclaughlin**
https://pulp.plan.io/issues/8233
closes #8233

### CODE REVIEW

File: pulp_ansible/app/tasks/collections.py [hunk 1]
```diff
@@ -717,6 +717,9 @@ async def _post_save(self, batch):
             docs_blob = d_content.extra_data.get("docs_blob", {})
             if docs_blob:
                 collection_version.docs_blob = docs_blob
+            execution_environment = d_content.extra_data.get("execution_environment", {})
```
Comment (fao89) [inline pulp_ansible/app/tasks/collections.py:720]: `execution_environment` needs to be set on `d_content.extra_data`
https://github.com/pulp/pulp_ansible/blob/master/pulp_ansible/app/tasks/collections.py#L397-L405

**Review by fao89 — COMMENTED** [review]
should this be on pulp-ansible 0.7? Or can it go on 0.8?

**Comment by bmclaughlin** [comment]
This should be on pulp-ansible 0.7 if possible.

File: pulp_ansible/app/tasks/collections.py (hunk 1 above)
Comment (bmclaughlin) [reply pulp_ansible/app/tasks/collections.py:720]: Thanks, thought I had removed this.  Seems ctrl-z wasn't my friend earlier today.

File: pulp_ansible/tests/functional/api/collection/v3/test_collection.py [hunk 2]
```diff
@@ -224,6 +224,7 @@ def test_collection_version(artifact, pulp_client, collection_detail):
     assert "files" in version
     assert "manifest" in version
     assert "requires_ansible" in version
+    assert "execution_environment" in version
```
Comment (bmbouter) [inline pulp_ansible/tests/functional/api/collection/v3/test_collection.py:227]: This is good, but we also need two more tests I think:

1) Upload a collection and assert that the API renders the data correctly
2) Sync in a collection and assert that the API renders the data correctly

Comment (bmbouter) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection.py:227]: For example here's a similar test added for uploading and then asserting on the `requires_ansible` field https://github.com/pulp/pulp_ansible/pull/466/files#diff-0ac20bc70d45017ec41ec2a189688b5814a44a91b0c55df22e03a3c9fd5c7b74R77

Comment (bmbouter) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection.py:227]: Here is one that does syncing and then asserting, also for the `requires_ansible` field: https://github.com/pulp/pulp_ansible/pull/466/files#diff-33689e25ac3c04bf435e438f2d0c86bbf6e21b4cc1524eeaf1e65cd33e8939f3R72-R91

Comment (bmbouter) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection.py:227]: Actually before making these tests, let me ask. How does data get into this field? In pulp_ansible it comes either from the artifact uploaded through the CLI or from sync. It can't come from an API call though.

Comment (bmclaughlin) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection.py:227]: I looked at adding both of those tests, but the rendered data would be `None` in either case (as the data is populated in Galaxy Importer by combining multiple files into said data).  I'm not sure how helpful that data point would be in tests?

I was leaning towards adding a Todo for myself to add additional tests (probably syncing in a collection) when execution environment data would be more widely available to test against.  Do you have a preference?

Comment (bmbouter) [reply pulp_ansible/tests/functional/api/collection/v3/test_collection.py:227]: The galaxy importer's data is returned here https://github.com/pulp/pulp_ansible/blob/8dd22a1ba89dfe2a11164d0272fc3e9fea67d3eb/pulp_ansible/app/tasks/collections.py#L153-L155 so that needs to be extended to populate the data. That would bring the data in during import. The unit test could then assert on that.

Then since the data is available on the server side (via the galaxy API V3) the Pulp sync code needs to bring it in to sync's clients. So the test we use there is upload data into repo a, make a distribution serving repo a, make repo b, make a distribution serving repo b, sync repo a-> b, and assert on the data now available in repo b. We have pulpToPulpV3 sync tests already so it's basically just doing that only with an upload at the beginning.

File: pulp_ansible/app/migrations/0031_collectionversion_execution_environment.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.2.18 on 2021-02-08 20:36
+
+import django.contrib.postgres.fields.jsonb
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('ansible', '0030_collectionversion_requires_ansible'),
```
Comment (fao89) [inline pulp_ansible/app/migrations/0031_collectionversion_execution_environment.py:10]:
```suggestion
        ('ansible', '0031_ansiblerepository_last_synced_metadata_time'),
```

Comment (fao89) [reply pulp_ansible/app/migrations/0031_collectionversion_execution_environment.py:10]: and please rename the migration to: 
```
0032_collectionversion_execution_environment.py 
```

File: pulp_ansible/app/migrations/0032_collectionversion_execution_environment.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# Generated by Django 2.2.18 on 2021-02-08 20:36
+
+import django.contrib.postgres.fields.jsonb
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('ansible', '0030_collectionversion_execution_environment'),
```
Comment (fao89) [inline pulp_ansible/app/migrations/0032_collectionversion_execution_environment.py:10]: it depends on the 31
```suggestion
        ('ansible', '0031_ansiblerepository_last_synced_metadata_time'),
```

Comment (bmclaughlin) [reply pulp_ansible/app/migrations/0032_collectionversion_execution_environment.py:10]: :facepalm:

**Comment by bmclaughlin** [comment]
Closing due to feature no longer being needed.
